lib/std/os/linux.zig

```diff
@@ -470,6 +469,347 @@ pub const O = switch (native_arch) {
     else => @compileError("missing std.os.linux.O constants for this architecture"),
 };
 
+// The 64-bit `stat` definition used by the Linux kernel,
+// which is one of the following choices:
+// - struct stat   on 64-bit target.
+// - struct stat64 on a 32-bit target.
+// - struct stat64 on a 64-bit target.
+pub const Stat = switch (native_arch) {
```

Comment:
Good change that I've been meaning to do for a while. I would actually suggest extracting this into a separate PR if you feel like it.